src/cls/rgw/cls_rgw_types.h

```diff
@@ -506,11 +506,12 @@ struct rgw_bi_log_entry {
   uint16_t bilog_flags;
   string owner; /* only being set if it's a delete marker */
   string owner_display_name; /* only being set if it's a delete marker */
+  set<string> zones_trace;
```

Comment:
could you add a `rgw_zone_set` typedef for this and use it throughout?
```c++
using rgw_zone_set = std::set<std::string>;
```
this would enable a couple of future optimizations, like using a `basic_sstring` sized to fit uuids (36 + \n), or `boost::container::flat_set` to cut down on its allocations